apps/graphql/src/apps/itinerary/helpers/Itineraries.js

```diff
@@ -181,3 +186,55 @@ const sanitizeSegment = (segment: ?ApiRouteItem): Segment => {
     arrival: apiRouteItemToArrival(segment),
   };
 };
+
+const parseBagDimensions = (dimensions: ?BagDimensions): ?string => {
+  if (dimensions != null) {
+    const { width, height, length } = dimensions;
+    if (width != null && height != null && length != null) {
+      return `${width} x ${height} x ${length}`;
+    }
+  }
+  return null;
+};
+
+const parseBagWeight = (weight: ?number): ?string => {
+  if (weight == null) {
+    return null;
+  }
+  return `${weight} kg`;
+};
+
+export const getHoldBagOptions = (
+  prices: ?ApiBagsPrice,
+  currency: ?string,
+  bagWeight: ?number,
+  bagDimensions: ?BagDimensions,
+): ?Array<HoldBagOption> => {
+  /**
+   * NOTE: For now there is only 1-3 available and bookable quantity for hold bags.
+   * This value is defined by current data available on endpoints and will be extended later
+   * with more combinations, after new bags implementation will be ready for booking
+   * endpoint https://docs.kiwi.com/booking/ (planned for Q2 2019)
+   */
+  const possibleHoldBagQuantities = [1, 2, 3];
+  const options = [];
+  const dimensions = parseBagDimensions(bagDimensions);
+  const weight = parseBagWeight(bagWeight);
+  for (const quantity of possibleHoldBagQuantities) {
```

Comment:
Maybe `reduce` can be used here to avoid `break;`
```js
  const options = possibleHoldBagQuantities.reduce((acc, quantity) => {
    const priceAmount = prices?.[quantity.toString()];
    if (priceAmount == null) {
      return acc;
    }
    const option = {
      quantity,
      dimensions,
      weight,
      price: {
        currency: currency,
        amount: priceAmount,
      },
    };
    return [...acc, option];
  }, []);

```

Reply:
Good idea, thanks, looks definitely better.

Reply:
Updated